src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -133,17 +122,15 @@ public ReadOnlySpan<char> FileName
 
         // Windows never fails getting attributes, length, or time as that information comes back
         // with the native enumeration struct. As such we must not throw here.
-
-        public FileAttributes Attributes
-            // It would be hard to rationalize if the attributes change after our initial find.
-            => _initialAttributes | (_status.IsReadOnly(FullPath, continueOnError: true) ? FileAttributes.ReadOnly : 0);
-
+        public FileAttributes Attributes => _status.GetAttributes(FullPath, FileName);
```

Comment:
Is this somehow a change of behavior and if so, could it be considered breaking?

Reply:
This was incorrect behavior, and part of the original bug.

Before: we reused the original attributes and also added the read-only check. This meant that if the user modified the hidden attribute, it would not be returned by `Attributes`, not even after refreshing.

After: We always return both read-only and hidden's latest values.

Reply:
Ok, as long as this is in sync with FileSystemInfo, it should be fine. Thanks.